compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ReshapeFusion.cpp

```diff
@@ -256,10 +256,13 @@ static std::optional<SmallVector<Value>> fuseAttentionWithReshapeByExpansion(
 
   // Create a new `AttentionOp` that has the computed operands/indexing maps.
   TypeRange resultTypes = ValueRange(outputs).getTypes();
+  std::optional<Value> mask =
+      attentionOp.getMask() ? std::optional<Value>(expandedOpOperands.back())
+                            : std::nullopt;
```

Comment:
I'm not sure what is happening here. Why are we wrapping a Value inside a std::optional? Value, is nullable, which means we can treat nullptr as the "optional" part of it.

Reply:
Good question, I am too a bit confused. Opted for a simpler approach